css/ast/src/lib.rs

```diff
@@ -73,18 +82,47 @@ pub enum BaseSelector {
 #[ast_node]
 pub struct Property {
     pub span: Span,
-    pub id: Text,
+    pub name: Text,
     pub value: Value,
-    /// Includes `!`
+    /// The span includes `!`
     pub important: Option<Span>,
 }
 
 #[ast_node]
 pub enum Value {
-    #[tag("Text")]
-    Raw(Text),
     #[tag("ParenValue")]
     Paren(ParenValue),
+    #[tag("UnitValue")]
+    Unit(UnitValue),
+    #[tag("Number")]
+    Number(Number),
+    #[tag("HashValue")]
+    Hash(HashValue),
+}
+
+#[ast_node]
+pub struct HashValue {
+    /// Includes `#`
+    pub span: Span,
+    /// Does **not** include `#`
+    pub value: Text,
+}
+
+#[ast_node]
+pub struct UnitValue {
+    pub span: Span,
+    pub value: Text,
+    pub unit: SpannedUnit,
+}
+
+#[ast_node]
+pub struct SpannedUnit {
+    pub span: Span,
+    pub unit: Unit,
+}
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum Unit {
+    Px,
 }
```

Comment:
Do we need to parse each unit? I thought we just need the unit in string?

- grass parses it since it needs to use it for transformation https://github.com/connorskees/grass/blob/master/src/unit/mod.rs#L13
- rust-cssparser serialize it as string https://github.com/servo/rust-cssparser/blob/master/src/tokenizer.rs#L94